driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -7,177 +7,144 @@ use crate::{
     models::{AccountState, Order},
     orderbook::StableXOrderBookReading,
 };
-use anyhow::{anyhow, bail, ensure, Result};
+use anyhow::{anyhow, bail, Result};
 use block_timestamp_reading::{BlockTimestampReading, CachedBlockTimestampReader};
 use ethcontract::{contract::Event, BlockNumber, H256};
-use futures::{
-    channel::oneshot, compat::Future01CompatExt as _, future::FutureExt, pin_mut, select_biased,
-};
+use futures::compat::Future01CompatExt as _;
 use orderbook::Orderbook;
 use std::collections::HashSet;
-use std::sync::{
-    atomic::{AtomicBool, Ordering},
-    Arc, Mutex,
-};
-use std::{process, thread, time::Duration};
+use std::sync::{Arc, Mutex};
 
 /// An event based orderbook that automatically updates itself with new events from the contract.
-#[derive(Debug)]
 pub struct UpdatingOrderbook {
-    orderbook: Arc<Mutex<Orderbook>>,
-    // Indicates whether the background thread has caught up with past events at which point the
-    // orderbook is ready to be read.
-    orderbook_ready: Arc<AtomicBool>,
-    // When this struct is dropped this sender will be dropped which makes the updater thread stop.
-    _exit_tx: oneshot::Sender<()>,
+    contract: Arc<dyn StableXContract + Send + Sync>,
+    web3: Web3,
+    /// We need a mutex because otherwise the struct wouldn't be Sync which is needed because we use
+    /// the orderbook in multiple threads. The mutex is locked in `get_auction_data` while the
+    /// orderbook is updated with new events.
+    behind_mutex: Mutex<BehindMutex>,
 }
 
-impl UpdatingOrderbook {
-    pub fn new(contract: Arc<dyn StableXContract + Send + Sync>, web3: Web3) -> Self {
-        let orderbook = Arc::new(Mutex::new(Orderbook::default()));
-        let orderbook_clone = orderbook.clone();
-        let orderbook_ready = Arc::new(AtomicBool::new(false));
-        let orderbook_ready_clone = orderbook_ready.clone();
-        let (exit_tx, exit_rx) = oneshot::channel();
+struct BehindMutex {
```

Comment:
naming nit: Is this kind of naming standard? I personally find it a bit weird. Can we call it UpdatingOrderbookInner or something a like?

Reply:
I don't know what is standard for this. Imo BehindMutex is more meaningful than ...Inner but weak opinion.

Reply:
I have usually seen it as `...Inner` or `...Context`.